### ISSUE TITLE
Including related messages breaks threading

### ISSUE THREAD

**Issue by tdavis** [djcb/mu#1195]
When "include related messages" is enabled (the new default per #1194), unrelated messages get put under preceding threads. Consider the following:

```
01:31:14 PM  Kelly Han (Confluence) + [Confluence] Kelly Han mentioned you...
01:31:14 PM  Kelly Han (Confluence)  \ [Confluence] Kelly Han mentioned you...
02/12/2018   NextDraft               \ Trust Nobody
```

The first two messages are related; they're actually the same since one is in the mailbox's inbox and the other from the archive folder. However, the newsletter under them is definitely not related -- they're not even sent to the same address. 

Messages beneath a thread will be "sucked" into the preceding thread until a new thread starts. For instance, I have some threads with four or more unrelated messages just because they happen to all be single messages without duplicate or related messages.

Disabling "include related messages" with `W` restores proper threading. I didn't notice a difference with fancy threading icons enabled. I'm not sure if this bug (intended behavior?) has always been the case; until 1.0 I didn't even know including related messages was an option ;)

Using mu 1.1, Emacs 25.3.1, Arch Linux

**Comment by djcb** [comment]
I think this is at the cosmetical level; the `\` means "first child", so they are no related, tho because the `\` are indented a bit they seem that way. 

Emacs doesn't really align these things very nicely... you can try to experiment with `mu4e-headers-*-mark` and see if there's something that fits better with your fonts.

**Comment by djcb** [comment]
BTW, the threading display changed a bit, does it work now for you?

**Comment by tdavis** [comment]
I just installed `mu` from git and it does seem to have been fixed; threading works properly with or without related messages being on. And I like the cute arrows! Although since I stopped indexing anything beyond `INBOX`, there aren't any "related" messages left to muck things up...

I'm happy to call this resolved if you'd like, I can always open it again if I notice the behaviour come back when related messages are present. Thanks for the update!
